### PR TITLE
[SPARK-34962][SQL] Explicit representation of * in UpdateAction and InsertAction in MergeIntoTable

### PR DESCRIPTION
**Pull request by tdas**
### What changes were proposed in this pull request?
Change UpdateAction and InsertAction of MergeIntoTable to explicitly represent star,

### Why are the changes needed?
Currently, UpdateAction and InsertAction in the MergeIntoTable implicitly represent `update set *` and `insert *` with empty assignments. That means there is no way to differentiate between the representations of "update all columns" and "update no columns". For SQL MERGE queries, this inability does not matter because the SQL MERGE grammar that generated the MergeIntoTable plan does not allow "update no columns". However, other ways of generating the MergeIntoTable plan may not have that limitation, and may want to allow specifying "update no columns". For example, in the Delta Lake project we provide a type-safe Scala API for Merge, where it is perfectly valid to produce a Merge query with an update clause but no update assignments. Currently, we cannot use MergeIntoTable to represent this plan, thus complicating the generation, and resolution of merge query from scala API.

Side note: fixed another bug where a merge plan with star and no other expressions with unresolved attributes (e.g. all non-optional predicates are `literal(true)`), then resolution will be skipped and star wont expanded. added test for that.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing unit tests

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala [hunk 1]
```diff
@@ -1309,6 +1310,28 @@ class PlanResolutionSuite extends AnalysisTest {
           case other => fail("Expect MergeIntoTable, but got:\n" + other.treeString)
         }
 
+        // merge with star should get resolved into specific actions even if there
```
Comment (tdas) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala:1313]: this is a test for for another bug i fixed where a merge plan with star and no other expressions with unresolved attributes (e.g. all non-optional predicates are `literal(true)`), then resolution will be skipped and star wont expanded. added test for that.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala [hunk 2]
```diff
@@ -404,8 +404,17 @@ case class MergeIntoTable(
     mergeCondition: Expression,
     matchedActions: Seq[MergeAction],
     notMatchedActions: Seq[MergeAction]) extends BinaryCommand with SupportsSubquery {
+
+  override lazy val resolved: Boolean = {
```
Comment (tdas) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala:408]: this fixes another bug where a merge plan with star and no other expressions with unresolved attributes (e.g. all non-optional predicates are `literal(true)`), then the resolution will be skipped and the star won't be expanded. added test for that.

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala:407]: How about we simply mark `InsertStarAction` and `UpdateStarAction` as unresolved as they must be expanded during analysis? Similar to `Star`

**Review by cloud-fan — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala (hunk 2 above)
Comment (tdas) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala:407]: oh yeah. thats much better!

**Review by cloud-fan — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
cc @aokolnychyi

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**